ArcGISRuntimeSDKQt_CppSamples/Analysis/Geotriggers/Geotriggers.cpp

```diff
@@ -228,23 +228,25 @@ void Geotriggers::getFeatureInformation(const QString& sectionName)
   }
 
   emit displayInfoChanged();
-
-  AttachmentListModel* featureAttachments = feature->attachments();
-
-  // Fetch attachments will automatically trigger upon instantiation of the AttachmentListModel
-  connect(featureAttachments, &AttachmentListModel::fetchAttachmentsCompleted, this, [this, sectionName](const QUuid&, const QList<Attachment*>& attachments)
-  {
-    if (attachments.isEmpty())
-      return;
-
-    // Get the first (and only) attachment for the feature
-    Attachment* sectionImageAttachment = attachments.first();
-
-    sectionImageAttachment->fetchDataAsync().then(this, [this, sectionImageAttachment, sectionName](const QByteArray&)
-    {
-      m_featureAttachmentImageUrls[sectionName] = sectionImageAttachment->attachmentUrl();
-      m_currentFeatureImageUrl = m_featureAttachmentImageUrls[sectionName];
-      emit displayInfoChanged();
-    });
-  });
+  // enableAutoFetch and enableAutoApplyEdits for AttachmentListModel are set as false
+  // to avoid automatic behavior. We will call fetchDataAsync explicitly as needed.
+  feature->attachments(false, false)->fetchAttachmentsAsync().then(
+      [this, sectionName](const QList<Attachment*>& attachments)
+      {
+        if (attachments.isEmpty())
+          return;
+
+        // Get the first (and only) attachment for the feature
+        Attachment* sectionImageAttachment = attachments.first();
+
+        sectionImageAttachment->fetchDataAsync().then(this, [this, sectionImageAttachment, sectionName](const QByteArray&)
+        {
+          m_featureAttachmentImageUrls[sectionName] = sectionImageAttachment->attachmentUrl();
+          m_currentFeatureImageUrl = m_featureAttachmentImageUrls[sectionName];
+          emit displayInfoChanged();
+        });
+        qDeleteAll(attachments);
+      });
+   auto future = m_gardenSections->applyEditsAsync(this);
+   Q_UNUSED(future)
```

Comment:
I don't know if you want to delete the attachments. I meant to delete the results from `applyEditsAsync`, which returns `QList<FeatureEditResult*>`.

Reply:
Sorry for that. Deleted `QList<FeatureEditResult*> results`.